quixstreams/sinks/base/sink.py

```diff
@@ -0,0 +1,80 @@
+import abc
+import logging
+from typing import Any, Dict, List, Tuple
+
+from quixstreams.models import HeaderValue
+from quixstreams.sinks.base.batch import SinkBatch
+
+logger = logging.getLogger(__name__)
+
+
+class BaseSink(abc.ABC):
+    # TODO: Docs
+
+    @abc.abstractmethod
+    def flush(self, topic: str, partition: int): ...
+
+    @abc.abstractmethod
+    def add(
+        self,
+        value: Any,
+        key: Any,
+        timestamp: int,
+        headers: List[Tuple[str, HeaderValue]],
+        topic: str,
+        partition: int,
+        offset: int,
+    ): ...
+
+    def on_paused(self, topic: str, partition: int): ...
+
+
+class BatchingSink(BaseSink):
+    # TODO: Docs (including "how to create your own sink)
+    _batches: Dict[Tuple[str, int], SinkBatch]
+
+    def __init__(self):
+        self._batches = {}
+
+    def __repr__(self):
+        return f"<BatchingSink: {self.__class__.__name__}>"
+
+    @abc.abstractmethod
+    def write(self, batch: SinkBatch): ...
+
+    def add(
+        self,
+        value: Any,
+        key: Any,
+        timestamp: int,
+        headers: List[Tuple[str, HeaderValue]],
+        topic: str,
+        partition: int,
+        offset: int,
+    ):
+        tp = (topic, partition)
+        batch = self._batches.get(tp)
+        if batch is None:
+            batch = SinkBatch(topic=topic, partition=partition)
+            self._batches[tp] = batch
+        batch.append(
+            value=value, key=key, timestamp=timestamp, headers=headers, offset=offset
+        )
+
+    def flush(self, topic: str, partition: int):
+        batch = self._batches.get((topic, partition))
+        if batch is not None:
```

Comment:
Small nitpick but this is more readable imo, less indentation level further on in the method.
```
if batch is None:
  return
```